configuration/src/main/java/io/airlift/configuration/ConditionalModule.java

```diff
@@ -49,26 +50,51 @@ public static <T> Module conditionalModule(Class<T> config, Predicate<T> predica
                 conditionalModule(config, predicate.negate(), otherwise));
     }
 
+    public static <T> Module conditionalModule(Class<T> config, String prefix, Predicate<T> predicate, Module module, Module otherwise)
+    {
+        return combine(
+                conditionalModule(config, prefix, predicate, module),
+                conditionalModule(config, prefix, predicate.negate(), otherwise));
+    }
+
     public static <T> Module conditionalModule(Class<T> config, Predicate<T> predicate, Module module)
     {
-        return new ConditionalModule<>(config, predicate, module);
+        return conditionalModule(config, Optional.empty(), predicate, module);
+    }
+
+    public static <T> Module conditionalModule(Class<T> config, String prefix, Predicate<T> predicate, Module module)
+    {
+        return conditionalModule(config, Optional.of(prefix), predicate, module);
+    }
+
+    public static <T> Module conditionalModule(Class<T> config, Optional<String> prefix, Predicate<T> predicate, Module module)
+    {
+        return new ConditionalModule<>(config, prefix, predicate, module);
     }
 
     private final Class<T> config;
+    private final Optional<String> prefix;
     private final Predicate<T> predicate;
     private final Module module;
 
-    private ConditionalModule(Class<T> config, Predicate<T> predicate, Module module)
+    private ConditionalModule(Class<T> config, Optional<String> prefix, Predicate<T> predicate, Module module)
     {
         this.config = requireNonNull(config, "config is null");
+        this.prefix = requireNonNull(prefix, "prefix is null");
         this.predicate = requireNonNull(predicate, "predicate is null");
         this.module = requireNonNull(module, "module is null");
     }
 
     @Override
     protected void setup(Binder binder)
     {
-        T configuration = buildConfigObject(config);
+        T configuration;
```

Comment:
I think this can be
```java
T configuration = prefix
        .map(value -> buildConfigObject(config, value))
        .orElseGet(() -> buildConfigObject(config));